core/modules/replica/Configuration.cc

```diff
@@ -39,79 +39,163 @@ namespace {
+    return params;
+}
 } // namespace
 
 namespace lsst {
 namespace qserv {
 namespace replica {
 
+
 // These (static) data members are allowed to be changed, and they are set
 // globally for an application (process).
 bool         Configuration::_databaseAllowReconnect = true;
 unsigned int Configuration::_databaseConnectTimeoutSec = 3600;
 unsigned int Configuration::_databaseMaxReconnects = 1;
 unsigned int Configuration::_databaseTransactionTimeoutSec = 3600;
-string       Configuration::_qservMasterDatabasePassword = "";
-string       Configuration::_qservWorkerDatabasePassword = "";
+string       Configuration::_qservCzarDbUrl = "mysql://qsreplica@localhost:3306/qservMeta";
+string       Configuration::_qservWorkerDbUrl = "mysql://qsreplica@localhost:3306/qservw_worker";
 bool         Configuration::_xrootdAllowReconnect = true;
 unsigned int Configuration::_xrootdConnectTimeoutSec = 3600;
+util::Mutex  Configuration::_classMtx;
 
 
-void Configuration::setQservMasterDatabasePassword(string const& newPassword) {
-    string result = newPassword;
-    swap(result, _qservMasterDatabasePassword);
+// ---------------
+// The static API.
+// ---------------
+
+void Configuration::setQservCzarDbUrl(string const& url) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    if (url.empty()) {
+        throw invalid_argument("Configuration::" + string(__func__) + "  empty string is not allowed.");
+    }
+    _qservCzarDbUrl = url;
 }
 
 
-void Configuration::setQservWorkerDatabasePassword(string const& newPassword) {
-    string result = newPassword;
-    swap(result, _qservWorkerDatabasePassword);
+string Configuration::qservCzarDbUrl() {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return _qservCzarDbUrl;
+}
+
+
+database::mysql::ConnectionParams Configuration::qservCzarDbParams(string const& database) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return connectionParams(_qservCzarDbUrl, database);
+}
+
+
+void Configuration::setQservWorkerDbUrl(string const& url) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    if (url.empty()) {
+        throw invalid_argument("Configuration::" + string(__func__) + "  empty string is not allowed.");
+    }
+    _qservWorkerDbUrl = url;
+}
+
+
+string Configuration::qservWorkerDbUrl() {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return _qservWorkerDbUrl;
+}
+
+
+database::mysql::ConnectionParams Configuration::qservWorkerDbParams(string const& database) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return connectionParams(_qservWorkerDbUrl, database);
 }
 
 
 void Configuration::setDatabaseAllowReconnect(bool value) {
-    swap(value, _databaseAllowReconnect);
+    util::Lock const lock(_classMtx, _context(__func__));
+    _databaseAllowReconnect = value;
+}
+
+
+bool Configuration::databaseAllowReconnect() {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return _databaseAllowReconnect;
```

Comment:
For simple data types you could just user `atomic<>` instead of a global mutex, mutex-level synchronization is probably an overkill in this case. Same is probably true for URL strings.

Reply:
I too was considering atomics. Then decided not to because I would still have to use the mutex for strings. I wish operations over strings in C++ were thread-safe.